main.go

```diff
@@ -315,6 +315,8 @@ func licenseHeader(path string, tmpl *template.Template, data licenseData) ([]by
 		// handle various cmake files
 		if base == "cmakelists.txt" || strings.HasSuffix(base, ".cmake.in") || strings.HasSuffix(base, ".cmake") {
 			lic, err = executeTemplate(tmpl, data, "", "# ", "")
+		} else if base == "Makefile" || base == "makefile" || strings.HasSuffix(base, ".mk") { // handle various make files
```

Comment:
cmake needs special support because it needs to look at multiple parts of the filename.  I think you can just add this to the case statement above, similar to Gemfile.